src/frontend/src/hooks/teams.hooks.ts

```diff
@@ -24,9 +25,9 @@ export interface CreateTeamPayload {
   isFinanceTeam: boolean;
 }
 
-export const useAllTeams = () => {
-  return useQuery<Team[], Error>(['teams'], async () => {
-    const { data } = await getAllTeams();
+export const useAllTeams = (onlyArchive: boolean) => {
```

Comment:
Same comment here about splitting things up or making the distinction when rendering